### PR TITLE
Added Get and Post request In api.js-HASEEB_KHALIL

### PR DESCRIPTION
**Pull request by Haseeb3927**
Added methods to get an energiser by id params, search query and also added a post method of adding energiser

## Naming Rules

Name your PR like this: ISSUENUMBER-TITLE-YOURNAME

## Description

Write a summary of your change. Explain why you have made it.
Added methods to get an energiser by id params, search query and also added a post method of adding energiser

## Related to

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #1

## Checklist:

- [* ] My code follows the style guidelines of this project
- [ * ] I have carefully reviewed my own code
- [ * ] I have commented my code
- [ * ] I have updated any documentation
Screenshot 2022-03-10 120851
Screenshot 2022-03-10 120826

### CODE REVIEW

File: server/api.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ require("dotenv").config();
 
 const router = Router();
 
+<<<<<<< Updated upstream
```
Comment (Luke-Rogerson) [inline server/api.js:6]: I think you might have commited this accidentally?

File: server/api.js [hunk 2]
```diff
@@ -30,9 +31,118 @@ router.get("/energisers", (req, res) => {
     }).catch((err) => {
         console.log(err);
         res.status(500).send(err);
+=======
```
Comment (Luke-Rogerson) [inline server/api.js:34]: And here

File: server/api.js [hunk 3]
```diff
@@ -30,9 +31,118 @@ router.get("/energisers", (req, res) => {
...
+router.post("/energisers", function (req, res) {
+  let energiserTitle = req.body.title;
+  let energiserImage = req.body.image;
+  let energiserDescription = req.body.description;
+  let energiserLink = req.body.link;
+
+//    || !energiserImage || !energiserLink
+
+  if (!energiserTitle  || !energiserDescription) {
+    return res
```
Comment (Luke-Rogerson) [inline server/api.js:52]: Good to see you thinking about validation

File: server/api.js [hunk 4]
```diff
@@ -30,9 +31,118 @@ router.get("/energisers", (req, res) => {
...
+    .catch((error) => {
+      console.error(error);
+      res.status(500).json(error);
+>>>>>>> Stashed changes
+    });
+});
+
+
+
+<<<<<<< Updated upstream
```
Comment (Luke-Rogerson) [inline server/api.js:143]: These have also been commited accidentally I think

File: server/api.js (hunk 3 above)
Comment (Haseeb3927) [reply server/api.js:37]: Thank you

File: server/api.js (hunk 2 above)
Comment (Haseeb3927) [reply server/api.js]: I'm not sure how did it get there 😄 
Shall I remove these ==== signs in VS code and commit again?

File: server/api.js [hunk 5]
```diff
@@ -3,64 +3,185 @@ require("dotenv").config();
...
+		pool
+			.query(
+				`SELECT id, title, description, link FROM energisers WHERE LOWER(title) LIKE LOWER('%${searchQuery}%') ORDER BY title`
+			)
+			.then((result) => res.json(result.rows))
+			.catch((error) => {
+				console.error(error);
+				res.status(500).json(error);
+			});
+	}
```
[the remark this answers is not in the document]
Comment (Haseeb3927) [reply server/api.js:85]: router.get("/energisers/search", function (req, res) {
	let searchQuery = req.query.term;


	if (searchQuery.length) {
		pool
			.query(
				`SELECT id, title, description, link FROM energisers WHERE LOWER(title) LIKE LOWER('%${searchQuery}%') ORDER BY title`
			)
			.then((result) => res.json(result.rows))
			.catch((error) => {
				console.error(error);
				res.status(500).json(error);
			});
	} else{
    res.send({ msg: `No Such Energisers found` });
  }
});


Is this a better idea?

File: server/api.js [hunk 6]
```diff
@@ -3,64 +3,185 @@ require("dotenv").config();
...
-    const description = req.body.description;
-    const link = req.body.link;
+//  ADD AN ENERGISER
+router.post("/energisers", function (req, res) {
+	const energiserTitle = req.body.title;
+	const energiserDescription = req.body.description;
+	const energiserLink = req.body.link;
 
+	// Basic Url testing. it should be starting with "http" or "https" following://
+	const validateUrl = /^(ftp|http|https):\/\/[^ "]+$/.test(energiserLink);
```
[the remark this answers is not in the document]
Comment (Haseeb3927) [reply server/api.js]: `ftp` deleted

**Pull request merged**